### PR TITLE
Graphql mutation to add daily stats added

### PR DESCRIPTION
**Pull request by achhetr**
you can submit form to add your daily stats uniquely in the status

### CODE REVIEW

File: app/graphql/mutations/add_daily_stat.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module Mutations
+  class AddDailyStat < BaseMutation
+    argument :date, String, required: true
+    argument :data, GraphQL::Types::JSON, required: true
+
+    field :daily_stat, Types::DailyStatType, null: false
+    field :errors, [String], null: false
+
+    def resolve(date:, data:)
```
Comment (saramic) [inline app/graphql/mutations/add_daily_stat.rb:9]: this is breaking the build - too much complexity in this single method https://app.circleci.com/pipelines/github/failure-driven/health-tracker/33/workflows/dcf7cd8d-bf92-4b07-a918-ac3fa6ddcb7b/jobs/35

File: app/javascript/react/pages/Home/Home.js [hunk 2]
```diff
@@ -19,7 +35,14 @@ const Home = () => {
   const addInputElementHandler = (e) => {
     e.preventDefault();
     if (nameInput || valueInput) {
```
Comment (saramic) [inline app/javascript/react/pages/Home/Home.js:37]: why is this needed?

File: app/javascript/react/pages/Home/Home.js [hunk 3]
```diff
@@ -19,7 +35,14 @@ const Home = () => {
...
-      // send item through graphql query
+      const data = {
```
Comment (saramic) [inline app/javascript/react/pages/Home/Home.js:38]: should we construct the data field like this? could we not simply render the existing data as key value pairs on the form and have the form update save all the key value's? so just have

```
const [dailyStatData, setDailyStatData] = useState({});

...

# set a date
# fetch daily stats for date
# render fields 
```

hmm maybe that is next up

then we wouldn't need the whole valueInput and nameInput but I'll give a pass on this maybe

**Review by saramic — COMMENTED** [review]
generally pretty good. take a look at my questions and see if you have some answers or code changes that may make sense. I am ok with the react form stuff although I thing I had a different idea, which I may try to get across better after this PR

File: app/graphql/mutations/add_daily_stat.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      daily_stat = DailyStat.find_by(date: Date.parse(date), user_id: current_user.id)
+      if daily_stat.nil?
+        daily_stat = current_user.daily_stats.new(date: Date.parse(date), data: data)
+      else
+        daily_stat = DailyStat.find_by(date: date, user_id: current_user.id)
+        current_data = daily_stat.data
+        data.each do |name, value|
+          current_data[name] = value
+        end
+        daily_stat.data = current_data
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/graphql/mutations/add_daily_stat.rb:21]: Yeah you are right, fixing this

File: app/graphql/mutations/add_daily_stat.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    field :errors, [String], null: false
+
+    def resolve(date:, data:)
+      current_user = context[:current_user]
+      # find previous stat or create new
+      daily_stat = DailyStat.find_by(date: Date.parse(date), user_id: current_user.id)
+      if daily_stat.nil?
+        daily_stat = current_user.daily_stats.new(date: Date.parse(date), data: data)
+      else
+        daily_stat = DailyStat.find_by(date: date, user_id: current_user.id)
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/graphql/mutations/add_daily_stat.rb:16]: agree

File: app/graphql/mutations/add_daily_stat.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    argument :data, GraphQL::Types::JSON, required: true
+
+    field :daily_stat, Types::DailyStatType, null: false
+    field :errors, [String], null: false
+
+    def resolve(date:, data:)
+      current_user = context[:current_user]
+      # find previous stat or create new
+      daily_stat = DailyStat.find_by(date: Date.parse(date), user_id: current_user.id)
+      if daily_stat.nil?
```
[the remark this answers is not in the document]
Comment (achhetr) [reply app/graphql/mutations/add_daily_stat.rb:13]: yes there is a method find_or_create_by

File: app/javascript/react/pages/Home/Home.js (hunk 3 above)
Comment (achhetr) [reply app/javascript/react/pages/Home/Home.js:38]: yes that was my next step, initial step was to create a POC to add or update value using GQL and form.

File: app/javascript/react/pages/Home/Home.js (hunk 2 above)
Comment (achhetr) [reply app/javascript/react/pages/Home/Home.js:37]: it was initially needed when button wasn't disabled, not needed now

File: app/graphql/mutations/add_daily_stat.rb (hunk 1 above)
Comment (achhetr) [reply app/graphql/mutations/add_daily_stat.rb:9]: function is lot smaller now, after next push build shouldn't fail

**Pull request merged**